File: app/src/main/java/org/d1scw0rld/bookbag/dto/Date.java
package org.d1scw0rld.bookbag.dto;

import android.support.annotation.NonNull;

import java.util.Locale;

public class Date
{
   public int iDay, 
              iMonth, 
              iYear; 

//   public Date()
//   {
//      iDay = 1;
//      iMonth = 1;
//      iYear = 1900;
//   }
   
   public Date(int iDay,
                int iMonth,
                int iYear)
   {
      this.iDay = iDay;
      this.iMonth = iMonth;
      this.iYear = iYear;
   }
   
   
   public Date(int iDate)
   {
      fromInt(iDate);
   }
   
//   public Date(Date oDate)
//   {
//      this(oDate.iDay, oDate.iMonth, oDate.iYear);
//   }
   
   public int toInt()
   {
      return iYear*10000 + iMonth*100 + iDay;
   }
   
   private void fromInt(int iDate)
   {
      iDay = iDate % 100;
      iMonth = (iDate / 100) % 100;
      iYear = iDate / 10000;
   }

   @NonNull
   @Override
   public String toString()
   {
      return String.format(Locale.getDefault(), "%02d", iDay) + "/" + String.format(Locale.getDefault(), "%02d", iMonth) + "/" + iYear;
   }
   
   @Override
   public boolean equals(Object o)
   {
      if (this == o) return true;

      if(iDay != ((Date)o).iDay) return false;
      if(iMonth != ((Date)o).iMonth) return false;
      return iYear == ((Date) o).iYear;
   }      

}

File: app/src/main/java/org/d1scw0rld/bookbag/dto/Date.kt
package org.d1scw0rld.bookbag.dto

import java.util.Locale

data class Date(
    val day: Int,
    val month: Int,
    val year: Int
) : Comparable<Date> {

    // Default constructor (sets day=1, month=1, year=1900 as in original)
    constructor() : this(1, 1, 1900)

    // Parse from an integer value (e.g., yyyymmdd)
    constructor(dateValue: Int) : this(
        day = dateValue % 100,
        month = (dateValue / 100) % 100,
        year = dateValue / 10000
    )

    // Copy constructor for compatibility
    constructor(otherDate: Date) : this(otherDate.day, otherDate.month, otherDate.year)

    fun toInt(): Int {
        return year * 10000 + month * 100 + day
    }

    override fun toString(): String {
        return String.format(Locale.getDefault(), "%02d/%02d/%d", day, month, year)
    }

    override fun compareTo(other: Date): Int {
        return toInt().compareTo(other.toInt())
    }
}
